src/protocol/http/org/apache/jmeter/protocol/http/curl/BasicCurlParser.java

```diff
@@ -252,4 +821,232 @@ public Request parse(String commandLine) {
+       }
+       URI uriProxy = null;
+       try {
+           uriProxy = new URI(proxyServerParameters);
+           request.setProxyServer("scheme", uriProxy.getScheme());
+           Optional<String> userInfoOptional = Optional.ofNullable(uriProxy.getUserInfo());
+           if (userInfoOptional.isPresent()) {
+               String userinfo = userInfoOptional.get();
+               if (userinfo.contains(":")) {
+                   String[] userInfo = userinfo.split(":");
+                   request.setProxyServer("username", userInfo[0]);
+                   request.setProxyServer("password", userInfo[1]);
+               }
+           }
+           Optional<String> hostOptional = Optional.ofNullable(uriProxy.getHost());
+           if (hostOptional.isPresent()) {
+               request.setProxyServer("servername", hostOptional.get());
+           }
+           if (uriProxy.getPort() != -1) {
+               request.setProxyServer("port", String.valueOf(uriProxy.getPort()));
+           } else {
+               request.setProxyServer("port", "1080");
+           }
+       } catch (URISyntaxException e) {
+           LOGGER.error("string '{}' cannot be converted to a URL", proxyServerParameters);
+           throw new IllegalArgumentException(proxyServerParameters + " cannot be converted to a URL");
+       }
+   }
+
+   /**
+    * Set the username and password of proxy server
+    *
+    * @param request               http request
+    * @param proxyServerUserPasswd the username and password of proxy server
+    */
+   private void setProxyServerUserInfo(Request request, String proxyServerUserPasswd) {
+       if (proxyServerUserPasswd.contains(":")) {
+           String[] userInfo = proxyServerUserPasswd.split(":");
+           request.setProxyServer("username", userInfo[0]);
+           request.setProxyServer("password", userInfo[1]);
+       }
+   }
+
+   /**
+    * Get post data by different type of data option
+    *
+    * @param postdata       the post data
+    * @param dataOptionName the different option of "--data"
+    * @return the post data
+    */
+   private String getPostDataByDifferentOption(String postdata, String dataOptionName) {
+       if (dataOptionName.equals("data-urlencode")) {
+           postdata = encodePostdata(postdata);
+       } else {
+           if (postdata.contains("@") && !dataOptionName.equals("data-raw")) {
+               postdata = postdata.replace("@", "");
+               postdata = readFromFile(postdata);
+               if (!dataOptionName.equals("data-binary")) {
+                   postdata = deleteLineBreak(postdata);
+               }
+           }
+       }
+       return postdata;
+   }
+
+   /**
+    * Encode the post data
+    *
+    * @param postdata the post data
+    * @return the result of encoding
+    *
+    */
+   private String encodePostdata(String postdata) {
+       String res = null;
+       if (postdata.contains("@")) {
+           String[] arr = postdata.split("@");
```

Comment:
What happens to the other parts after potentially more `@` chars in the string?

Reply:
If you start the data with the character @, the rest should be a filename.

Reply:
it will also encode the contents of the file

Reply:
Yes i have try it by proxy of Jmeter
1. Result of curl command and was caught by Jmeter Recording Controller: `foo1=foo1%3Daaa%26%0D%0A%0D%0Afoo2%3Db%0D%0A%0D%0Abbadd+a+value+in+this+file+`
2. Result of Http Requst in JMeter created by curl "curl 'https://postman-echo.com/post' -H 'Content-Type: application/x-www-form-urlencoded' -H 'cache-control: no-cache' --data-urlencode 'foo1@C:\Test\test.txt'" `foo1=foo1%3Daaa%26%0D%0A%0D%0Afoo2%3Db%0D%0A%0D%0Abbadd+a+value+in+this+file+`

Reply:
And where is a second `@` in the argument to `--data-urlencode` for your test?

Reply:
I have change split("@") to split("@",2), and if the augument contains a "@", this "@" will be encoded too. I have test it by JMeter recording.

Reply:
Maybe you want to add a junit test.